oss-cloud-backend/handler.js

```diff
@@ -16,28 +17,130 @@ module.exports.hello = async event => {
   };
 };
 
-
-// returns all pull requests from parents of all forked repos for a single github user
+// add a contributor to database if he does not already exist and is registered on GitHub
 // POST expected json
 // {
-//    username: string  
+//    username: string
+//    firstName: string
+//    lastName: string
 // }
-module.exports.getPulls = (event, context, callback) => {
-  var body = JSON.parse(event.body);
-  getPullsHandler.getForkedRepos(body.username) // get all repos of a user
-    .then(repos => getPullsHandler.getRepoDetails(repos))
-    .then(repos => getPullsHandler.getParentRepos(repos))
-    .then(repos => getPullsHandler.getUserPulls(body.username, repos))
-    .then(pulls => callback(null, pulls))
-    .catch(err => {
-      console.log("error in getPulls handler: ", err);
-      let response = {
-        statusCode: err.status,
+module.exports.addContributor = async (event, context, callback) => {
+  // check if request is valid
+  try {
+    // TODO create generic function for checking validity of a body
+    var body = JSON.parse(event.body);
+    // check if keys exist
+    body.username;
+    body.firstName;
+    body.lastName;
+    if (Object.keys(body).length !== 3) {
+      throw error;
```

Comment:
I don't see `error` defined

Reply:
It now throws "Invalid number of attributes in JSON".